### PR TITLE
Check for presence of channel in multichan license at install

### PR DESCRIPTION
**Pull request by pandemicsyn**
#### What this PR does / why we need it:

With the introduction of multichannel licenses we can't rely on the top level ChannelID of the license files anymore to validate whether a channel is allowed by a license.  

#### Which issue(s) this PR fixes:

https://app.shortcut.com/replicated/story/112555/installing-an-ec-release-from-a-non-default-release-channel-fails

#### Does this PR require a release note?

```release-note

```

### CODE REVIEW

File: cmd/embedded-cluster/install.go [hunk 1]
```diff
@@ -266,7 +266,7 @@ func getLicenseFromFilepath(licenseFile string) (*kotsv1beta1.License, error) {
 		// if the app is different, we will not be able to provide the correct vendor supplied charts and k0s overrides
 		return nil, fmt.Errorf("license app %s does not match binary app %s, please provide the correct license", license.Spec.AppSlug, rel.AppSlug)
 	}
-	if rel.ChannelID != license.Spec.ChannelID {
+	if !channelIDExists(license.Spec.Channels, rel.ChannelID) {
```
Comment (laverya) [inline cmd/embedded-cluster/install.go:269]: should we still handle pre-'channels' licenses? IIRC that field didn't always exist

Comment (laverya) [reply cmd/embedded-cluster/install.go:269]: (plausibly we shouldn't as licenses should come with the binary download, but...)

Comment (laverya) [reply cmd/embedded-cluster/install.go:269]: Well, our CI uses a license file like that 😄 

Make sure to update the error message below this too, and the unit testing

Comment (pandemicsyn) [reply cmd/embedded-cluster/install.go]: @laverya heh yea sorry this was a WIP (still a draft)

Comment (laverya) [reply cmd/embedded-cluster/install.go]: oh no this was just me going "oh hey PR from a username I don't recognize", no worries at all

File: cmd/embedded-cluster/install.go [hunk 2]
```diff
@@ -266,10 +266,10 @@ func getLicenseFromFilepath(licenseFile string) (*kotsv1beta1.License, error) {
 		// if the app is different, we will not be able to provide the correct vendor supplied charts and k0s overrides
 		return nil, fmt.Errorf("license app %s does not match binary app %s, please provide the correct license", license.Spec.AppSlug, rel.AppSlug)
 	}
-	if rel.ChannelID != license.Spec.ChannelID {
+	if !channelIDExists(license, rel.ChannelID) {
 		// if the channel is different, we will not be able to install the pinned vendor application version within kots
 		// this may result in an immediate k8s upgrade after installation, which is undesired
-		return nil, fmt.Errorf("license channel %s (%s) does not match binary channel %s, please provide the correct license", license.Spec.ChannelID, license.Spec.ChannelName, rel.ChannelID)
+		return nil, fmt.Errorf("license channel %s is not allowed by license, please provide the correct license", rel.ChannelID)
```
Comment (laverya) [inline cmd/embedded-cluster/install.go:272]:
```suggestion
		return nil, fmt.Errorf("license channel does not match binary channel %s, please provide the correct license", rel.ChannelID)
```

File: cmd/embedded-cluster/install.go (hunk 1 above)
Comment (pandemicsyn) [reply cmd/embedded-cluster/install.go]: ah cool, my work flow is little commit spammy 😁 but i appreciate the early looks! @laverya think this is ready to go if ya wanna take another peek.

**Review by divolgin — APPROVED** [review]

File: cmd/embedded-cluster/install.go [hunk 3]
```diff
@@ -266,10 +266,10 @@ func getLicenseFromFilepath(licenseFile string) (*kotsv1beta1.License, error) {
 		// if the app is different, we will not be able to provide the correct vendor supplied charts and k0s overrides
 		return nil, fmt.Errorf("license app %s does not match binary app %s, please provide the correct license", license.Spec.AppSlug, rel.AppSlug)
 	}
-	if rel.ChannelID != license.Spec.ChannelID {
+	if !channelIDExists(license, rel.ChannelID) {
 		// if the channel is different, we will not be able to install the pinned vendor application version within kots
 		// this may result in an immediate k8s upgrade after installation, which is undesired
-		return nil, fmt.Errorf("license channel %s (%s) does not match binary channel %s, please provide the correct license", license.Spec.ChannelID, license.Spec.ChannelName, rel.ChannelID)
+		return nil, fmt.Errorf("license channel does not match binary channel %s, please provide the correct license", rel.ChannelID)
```
Comment (laverya) [inline cmd/embedded-cluster/install.go:272]: I would still prefer a friendlier error message here, ideally including what channels the license DOES allow, and possibly adding the rel.ChannelSlug as a friendlier thing to look for if that's embedded in the binary (and I believe it is)

Comment (pandemicsyn) [reply cmd/embedded-cluster/install.go:272]: @laverya 👍 yea you're right. let me send along another update.

Comment (pandemicsyn) [reply cmd/embedded-cluster/install.go]: @laverya ok mind taking another peek ? https://github.com/replicatedhq/embedded-cluster/pull/1220/commits/6467fae350bee3a339d30ae12ccd2b799a81c6a3

**Review by laverya — APPROVED** [review]

**Comment by ajp-io** [comment]
FYI this PR should have had the release notes section filled out. I didn't realize we had fixed this and in what version. Looks like this was released in 1.15, so I'll add this to our release notes:
>Fixes an issue that prevented you from installing Embedded Cluster using a multi-channel license and a channel other than the license's default.

**Pull request merged**